---
layout: page
title: About Me
---


### <u>Publications</u>

- *"Application of a comprehensive platform technology for detection of cancer-associated genomic alterations and viral DNA in liquid biopsies"*. **Annals of Oncology**, 2018. ([Second Author](https://academic.oup.com/annonc/article/29/suppl_9/mdy441.010/5200219))     

- *"An enhanced next-generation sequencing liquid biopsy assay for simultaneous detection of somatic variants, microsatellite instability and viral DNA."*. **Journal of Clinical Oncology**, 2018. ([Second Author](http://ascopubs.org/doi/abs/10.1200/JCO.2018.36.15_suppl.e24107)) 

- *"LDSplitDB: A database for studies of meiotic recombination hotspots in MHC using human genomic data"*. **BMC Medical Genomics**, 2018. ([Second Author](https://bmcmedgenomics.biomedcentral.com/articles/10.1186/s12920-018-0351-0))      

- *"Cytofkit: A Bioconductor Package for an Integrated Mass Cytometry Data Analysis Pipeline"*. **PLoS Comput Biol**, 2016. ([First Author](http://journals.plos.org/ploscompbiol/article?id=info:doi/10.1371/journal.pcbi.1005112))

- *"flowAI: automatic and interactive anomaly discerning tools for flow cytometry"*. **Bioinformatics**, 2016. ([Second Author](http://bioinformatics.oxfordjournals.org/content/early/2016/05/06/bioinformatics.btw191))

- *"ARG-walker: Inference of Individual Specific Strengths of Meiotic Recombination Hotspots by Population Genomics Analysis"*. **BMC Genomics**, 2015. ([First Author](http://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-16-S12-S1))

- *"Optimization of production conditions for mushroom polysaccharides with high yield and antitumor activity"*. **Carbohydrate Polymers**, 2012. ([Third Author](https://www.researchgate.net/publication/256975016_Optimization_of_production_conditions_for_mushroom_polysaccharides_with_high_yield_and_antitumor_activity))


### <u>Patents</u>

- *"Method for Detection and Quantification of Genetic Alterations"*. **10201805450Y**, 2018.

- *"Methods for Identifying Clusters in a Dataset, Methods of Analyzing Cytometry Data with the Aid of a Computer and Methods of Detecting Cell Sub-populations in a Plurality of Cells"*. **US 2017-0371886 A1**, 2017.

- *"ModalGATE: An Efficient Gating Algorithm Facilitating Automated and Reproducible Analysis of Flow Cytometry Data"*. **SIGN/P/09834/00/SG**, 2017.



### <u>Developed Softwares</u>

&emsp; ***R packages*** 

- [cytofkit](http://watson.nci.nih.gov/bioc_mirror/packages/cytofkit/) An integrated analysis pipeline for high-dimensional and high-throughput mass cytometry data. (Ranked among top 20% on [Bioconductor](http://bioconductor.org/packages/stats/bioc/cytofkit/), with average 300 download times per month)
- [flowAI](https://bioconductor.org/packages/flowAI/) Automatic and interactive quality control for flow cytometry data. (Ranked among top 50% on [Bioconductor](http://bioconductor.org/packages/stats/bioc/flowAI/), with average 100 download times per month)
- [ClusterX](https://github.com/JinmiaoChenLab/ClusterX) A tool for visualizing the high-dimensional mass cytometry data as well as automatically detecting cell populations by fast density-based clustering, generally applicable to any multi-parametric data.
- [RphenoGraph](https://github.com/JinmiaoChenLab/Rphenograph) R implementation of PhenoGraph , a fast and accurate high-dimensional clustering algorithm.
- [uSORT](https://github.com/JinmiaoChenLab/uSORT) A self-refining ordering pipeline for gene selection.
- [StackedDensityPlot](https://github.com/haoeric/StackedDensityPlot) A flexible and easy-to-use function for building stacked density plot based on ggplot2.

&emsp; ***Shiny APP*** 

- [flowAIshinyAPP](https://chenhao.shinyapps.io/flowAIshinyAPP/) A web application for interactive quality control of flow cytometry data
- [cytofkitShinyAPP](https://chenhao.shinyapps.io/cytofkitShinyAPP/) A web application for interactive visualization and exploration of cytofkit analysis results

&emsp; ***Perl Tool*** 

- [ARGwalker](https://github.com/haoeric/ARGwalker) Ancestral Recombination Graph (ARG) mining based estimation of individual recombination strength
   


