---
title: "Encoding clustering"
author: "haoeric"
date: "9 October, 2014"
output:
  html_document:
    toc: yes
---

Introduction
===================

An encoding based high dimension data interpretation method, for Cytof, RNA-seq and any data presented with a high dimension matrix format. The method is drafted as below:

##### High dimension data clustering:
1. column(channel)-wise density based clustering and encoding to sequences
2. multi-sequence alignment and build the phylogenetic tree 
3. set threshold to cut the tree to clusters

##### Biological application:
4. build the cell differentiation path with MST
5. interprete the graph, direction, branch, cell fate
6. maching learning based (eg. random walk) cell fate estimation based on the 

Key Challenges
=====================
1. which method for density based clustering, one dimension or two (channel number)?
2. how to encoding the density based clusters?
3. how to cluster the phylogenetic tree?
4. how to determaine the distance of clusters, and transform to MST?
5. how to determain the direction of cell differentiaion?
6. cluster annotation and interpretation, cell differentiation path explanation
7. which ML methods used for cell fate prediction?

Main Procedure
======================
### 1. Density based channel data encoding

The key drawback of the pupular density based cluster methods like **DBSCAN** and **OPTICS** is that they expect some kind of density drop to detect cluster borders. Moreover, they cannot detect intrinsic cluster structures which are prevalent in the majority of real life data. A variation of DBSCAN, **EnDBSCAN**, efficiently detects such kinds of structures. On data sets with, for example, overlapping Gaussian distributions - a common use case in artificial data - the cluster borders produced by these algorithms will often look arbitrary, because the cluster density decreases continuously. On a data set consisting of mixtures of Gaussians, these algorithms are nearly always outperformed by methods such as **EM clustering** that are able to precisely model this kind of data.


**Package ‘EMCluster’**

```{r}
library(EMCluster, quiet = TRUE)
set.seed(1234)
x1 <- da1$da
plot(x1)

emobj <- simple.init(x1, nclass = 9)
emobj <- shortemcluster(x1, emobj)
summary(emobj)

ret <- emcluster(x1, emobj, assign.class = TRUE)
summary(ret)

x1$cluster <- as.factor(ret$class)
library(ggplot2)
ggplot(x1, aes(x=x, y=y, color = cluster)) + geom_point()

```

These utility functions are to convert matrices in different formats:

LTSigma2variance(x)
variance2LTSigma(x)
LTsigma2var(x1, p = NULL)
var2LTsigma(x1)
class2Gamma(class)
Gamma2class(Gamma)

```{r}
library(EMCluster, quiet = TRUE)
x <- da2$LTSigma
class <- da2$class
y <- LTSigma2variance(x)
str(y)
y <- variance2LTSigma(y)
str(y)
sum(x != y)
Gamma <- class2Gamma(class)
class.new <- Gamma2class(Gamma)
sum(class != class.new)
```

### 2. Multiple Alignment and Phylogenetic trees based clustering

score matrix

http://a-little-book-of-r-for-bioinformatics.readthedocs.org/en/latest/src/chapter5.html



### 3. Dendrogram based MST buiding



### 4. Cell differentiation interpretation